src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -1418,6 +1418,102 @@ def aks_upgrade(cmd, client, resource_group_name, name, kubernetes_version, no_w
     return sdk_no_wait(no_wait, client.create_or_update, resource_group_name, name, instance)
 
 
+def aks_use_dev_spaces(cmd, client, cluster_name, resource_group_name, space_name='default', parent_space_name=None):  # pylint: disable=line-too-long
+    """
+    Use Azure Dev Spaces with a managed Kubernetes cluster.
+
+    :param cluster_name: Name of the managed cluster.
+    :type cluster_name: String
+    :param resource_group_name: Name of resource group. You can configure the default group. \
+    Using 'az configure --defaults group=<name>'.
+    :type resource_group_name: String
+    :param space_name: Name of the dev space to use.
+    :type space_name: String
+    :param parent_space_name: Name of a parent dev space to inherit from when creating a new dev space. \
+    By default, if there is already a single dev space with no parent, the new space inherits from this one.
+    :type parent_space_name: String
+    """
+
+    dev_spaces_extension = 'dev-spaces-preview'
+
+    should_create_resource = False
+    if not _is_dev_spaces_extension_installed(dev_spaces_extension):
+        if _install_dev_spaces_extension(dev_spaces_extension):
+            should_create_resource = True
+    else:
+        should_create_resource = True
+
+    if should_create_resource:
+        try:
+            # Adding the installed extension in the path
+            from azure.cli.core.extension import get_extension_path
+            ext_dir = get_extension_path(dev_spaces_extension)
+            sys.path.append(ext_dir)
+            # Import the extension module
+            from importlib import import_module
+            azext_custom = import_module('azext_dev_spaces_preview.custom')
+            azext_custom.ads_use_dev_spaces(
+                cluster_name, resource_group_name,
+                space_name, parent_space_name)
```

Comment:
What is this for? This introduces an implicit dependency from the CLI to the extension module. This is significant change we can't ignore.